### PR TITLE
Improve open_mfdataset deprecation warnings

### LINKED ISSUE #3091: auto_combine warnings with open_mfdataset
I'm seeing this warning when using `open_mfdataset(list_of_files)` on master.
```
/gpfs/u/home/dcherian/python/xarray/xarray/backends/api.py:783: FutureWarning: In xarray version 0.13 `auto_combine` will be deprecated.
/gpfs/u/home/dcherian/python/xarray/xarray/backends/api.py:783: FutureWarning: The datasets supplied have global dimension coordinates. You may want
to use the new `combine_by_coords` function (or the
`combine='by_coords'` option to `open_mfdataset` to order the datasets
before concatenation. Alternatively, to continue concatenating based
on the order the datasets are supplied in in future, please use the
new `combine_nested` function (or the `combine='nested'` option to
open_mfdataset).
```

I don't think we should be issuing this warning when using `open_mfdataset` with default kwargs. Also there's a typo: "in in future"

### PR DESCRIPTION
**Pull request by TomNicholas**
- [x] Closes #3091
 - [x] Tests added

### CODE REVIEW

**Comment by dcherian** [comment]
Awesome. I think this looks great. Sorry for not catching it earlier.

**Comment by max-sixty** [comment]
That's perfect! Thanks @TomNicholas

File: xarray/core/combine.py [hunk 1]
```diff
@@ -523,7 +523,8 @@ def combine_by_coords(datasets, compat='no_conflicts', data_vars='all',
 
 
 def auto_combine(datasets, concat_dim='_not_supplied', compat='no_conflicts',
-                 data_vars='all', coords='different', fill_value=dtypes.NA):
+                 data_vars='all', coords='different', fill_value=dtypes.NA,
```
Comment (shoyer) [inline xarray/core/combine.py]: Don't worry about it here -- it's not worth the trouble for a deprecated function -- but in the future when you want something like this the clean way to do it is probably to make a private helper function `_auto_combine` that adds the extra non-user-facing arguments.

**Comment by shoyer** [comment]
thanks @TomNicholas !

**Pull request merged**